modules/nw-operator-vsphere-multiple-nic.adoc

```diff
@@ -0,0 +1,75 @@
+
+//Bare metal only?
+[IMPORTANT]
+====
+Each NIC must have access to link:https://quay.io/repository/[Quay.io] to obtain packages that are required for the installation of your cluster.
+====
+
+If you do not configure internal and external CIDR implementations in the {vmw-short} CCM configuration, the {vmw-short} CCM selects the first NIC that the host operating system reports as the primary network interface. This configuration can cause new nodes that associate with a `MachineSet` object with a single NIC to become unusable as each new node receives the `node.cloudprovider.kubernetes.io/uninitialized` taint. In certain situations, the reported network interface might not be the expected network interface. These situations can cause communication issues with the Kubernetes API server that can cause installation of the cluster to fail.
+
+[IMPORTANT]
+====
+If you configured multiple NICs for a network implementation but intend to use only a single NIC for your current needs, ensure that you specify the same IP address for the internal and external CIDR implementations. This configuration ensures that the CCM selects your preferred NIC and not the first NIC that the host operating system reports.
+====
+
+.Example error message because of the {vmw-short} CCM selecting an incompatible NIC
+[source, text]
+----
+ERROR Bootstrap failed to complete: timed out waiting for the condition
+ERROR Failed to wait for bootstrapping to complete. This error usually happens when there is a problem with control plane hosts that prevents the control plane operators from creating the control plane.
+----
+
+.Prerequisites
+
+* You created Kubernetes manifest files for your {product-title} cluster.
+
+.Procedure
+
+. From the directory where you store your {product-title} cluster manifest files, open the `manifests/cluster-infrastructure-02-config.yml` manifest file.
+
+. Add a `nodeNetworking` object to the file and specify internal and external network subnet CIDR implementations for the object.
++
+.Example of a configured `cluster-infrastructure-02-config.yml` manifest file
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1
+kind: Infrastructure
+metadata:
+  name: cluster
+spec:
+  cloudConfig:
+    key: config
+    name: cloud-provider-config
+  platformSpec:
+    type: VSphere
+    vsphere:
+      failureDomains:
+      - name: generated-failure-domain
+      ...
+      nodeNetworking:
+        external: <1>
+          networkSubnetCidr: 192.168.100.0/24
+        internal: <2>
+          networkSubnetCidr: 172.16.0.0/24
```

Comment:
I think the Cidrs should be the same and they should be the MachineNetwork.

We need to provide clear guidance for every Cidr we use.  This came up in https://github.com/openshift/openshift-docs/pull/76001 as well. 

If it's just the MachineNetwork CIDR, then. we should just say so, especially since this is at install time and the customer has `install-config.yaml` on hand.

Reply:
Don't forget IPv6.  Are these fields modifiable when converting to Dual-Stack?

We need to stop assuming single-stack IPv4.

Reply:
@elmiko Do you know if these fields support dual stack? Do they even need to or will CCM select the IPs from the same interface as long as you provide one of the CIDRs?

Reply:
We have to inject `"fd69::2/128"` into `excludeNetworkSubnetCidr`  for dual-stack due to OVN-K stuff.

Reply:
https://github.com/openshift/cluster-cloud-controller-manager-operator/blob/release-4.16/pkg/cloud/vsphere/vsphere_config_transformer.go#L153

Reply:
> Do you know if these fields support dual stack?

it appears that they do, the configuration fields for the internal/external and exclude internal/external al lgo through this logic: https://github.com/openshift/cloud-provider-vsphere/blob/master/pkg/cloudprovider/vsphere/nodemanager.go#L255-L262

which ultimately results in the values being parsed by https://pkg.go.dev/net#ParseCIDR . so, i would expect all those fields to be able to handle ip4 and ipv6 addresses as specific in the exclude examples eg `"192.0.2.0/24,fe80::1/128"`.

Reply:
> Do they even need to or will CCM select the IPs from the same interface as long as you provide one of the CIDRs?

this i'm not sure about, i would need to study the code more to understand how that works.

Reply:
Okay, as long as we can specify both I'm not too concerned about it. It looks like the syntax then is not a YAML list, but a comma-separated string, correct?

Reply:
the CRD is `[]string` as indicated below.     The `cloud-conf.yaml` that is generated is comma list.